### PR TITLE
Add test case for JSON resource marshaling 

### PR DESCRIPTION
**Pull request by rahulkarmore**
Hi creator,
Added test cases for JSON Resources :
CommitResult
Helps: https://github.com/google/go-github/issues/55

I have followed all the guidelines to contribute to this repository.
Please revert in case of any changes are needed.
I am a first-time contributor to Hackoctober, It will be a great help if u can encourage me & accept my contribution.

### CODE REVIEW

File: github/search_test.go [hunk 1]
```diff
@@ -914,3 +914,28 @@ func TestLabelsSearchResult_Marshal(t *testing.T) {
 
 	testJSONMarshal(t, u, want)
 }
+func TestCommitResult_Marshal(t *testing.T) {
```
Comment (gmlewis) [inline github/search_test.go:917]: Please add a blank line before this test function.

**Review by gmlewis — CHANGES_REQUESTED** [review]

File: github/search_test.go (hunk 1 above)
Comment (rahulkarmore) [reply github/search_test.go:918]: added a blank line, Thanks for suggesting changes.

File: github/search_test.go [hunk 2]
```diff
@@ -914,3 +914,42 @@ func TestLabelsSearchResult_Marshal(t *testing.T) {
...
+		SHA:         String("test"),
+		HTMLURL:     String("hurl"),
+		CommentsURL: String("curl"),
+		URL:         String("url"),
+		Repository:  &Repository{ID: Int64(1)},
+		Score:       Float64(123),
+		Commit:      &Commit{SHA: String("test")},
+		Author:      &User{ID: Int64(1)},
+		Committer:   &User{ID: Int64(1)},
+		Parents:     []*Commit{},
```
Comment (gmlewis) [inline github/search_test.go:931]: Any reason for not supplying a single parent?

Comment (gmlewis) [reply github/search_test.go:931]: Never mind, let's get this in before I go OOO.

**Review by gmlewis — APPROVED** [review]

File: github/search_test.go (hunk 2 above)
Comment (rahulkarmore) [reply github/search_test.go:931]: it's just an omitempty so I want to try to ignore this. But Thanks @gmlewis for approving :)

**Pull request merged**